libbeat/api/make_listener_posix.go

```diff
@@ -0,0 +1,83 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//+build !windows
+
+package api
+
+import (
+	"fmt"
+	"net"
+	"os"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/api/npipe"
+)
+
+func makeListener(cfg Config) (net.Listener, error) {
+	if len(cfg.User) > 0 {
+		return nil, errors.New("specifying a user is not supported under this platform")
+	}
+
+	if len(cfg.SecurityDescriptor) > 0 {
+		return nil, errors.New("security_descriptor option for the HTTP endpoint only work on Windows")
+	}
+
+	if npipe.IsNPipe(cfg.Host) {
+		return nil, fmt.Errorf(
+			"cannot use %s as the host, named pipes are only supported on Windows",
+			cfg.Host,
+		)
+	}
+
+	network, path, err := parse(cfg.Host, cfg.Port)
+	if err != nil {
+		return nil, err
+	}
+
+	if network == "unix" {
+		if _, err := os.Stat(path); !os.IsNotExist(err) {
```

Comment:
The double negative if predicate, `!os.IsNotExist(err)` is a bit hard to parse. Any reason we couldn't do `os.IsExist(err)`?

Reply:
Just chatted with @ph off-PR about this change so wanted to summarize our discussion here. Apparently, if the double-negative check here is replaced with `os.IsExist(err)`, things don't work as expected:

```
$ touch /tmp/foo
$ ./filebeat -c filebeat.dev.yml -E http.host=unix:///tmp/foo                                                                                                                                      
Exiting: could not start the HTTP server for the API: listen unix /tmp/foo: bind: address already in use
```

But if we keep the double-negative check, everything works as expected. So we are bringing it back.